exporter/otlp/lib/opentelemetry/exporter/otlp/exporter.rb

```diff
@@ -50,7 +50,6 @@ def initialize(endpoint: config_opt('OTEL_EXPORTER_OTLP_TRACES_ENDPOINT', 'OTEL_
                        metrics_reporter: nil)
           raise ArgumentError, "invalid url for OTLP::Exporter #{endpoint}" if invalid_url?(endpoint)
           raise ArgumentError, "unsupported compression key #{compression}" unless compression.nil? || compression == 'gzip'
-          raise ArgumentError, 'headers must be comma-separated k=v pairs or a Hash' unless valid_headers?(headers)
```

Comment:
Do we want to lose this validation? Does `parse_headers` have any failure cases that could `raise`? If so, will the error be obvious?

Reply:
Looking at the other SDK implementations, there wasn't any validation there as far as I could tell, so I had mixed feelings about keeping the validation. E.g. It looks like blank or invalid values are omitted but do not raise errors. 

Do you feel strongly that we should fail-fast here?

Also, I may have missed some test cases that could reveal some issues with this PR, e.g. Invalid Byte Sequence. I will add some for those I could think of.

Reply:
The spec says:

> The API or SDK may _fail fast_ and cause the application to fail on initialization, e.g. because of a bad user config or environment

We've generally tried to be helpfully verbose with initialization errors rather than silently ignoring config problems. I'd like to retain that, if possible. We don't *have* to, though.

Reply:
I have mixed feelings about failing-fast in production environments because that seems to run contrary to the experience with other components like instrumentation, which intentionally ignores errors to not interfere with normal application use. 

Part of me wants to guide our users to write a custom error handler that would "fail fast", giving them some control of what to ignore and what to address, however that could be daunting for them in production environments when sifting through errors. 

That lead me down the path of reporting errors using `OpenTelemetry.handle_error`. If it is indeed our preference to Fail-Fast when it comes to Exporter configurations then I will go back to the fail-fast approach and raise `ArgumentError` instead.

Reply:
> I have mixed feelings about failing-fast in production environments because that seems to run contrary to the experience with other components like instrumentation, which intentionally ignores errors to not interfere with normal application use.

The choice to ignore errors in instrumentation [aligns with the spec](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/error-handling.md#basic-error-handling-principles):

> ..., but MUST NOT cause the application to fail later at run time

Initialization is a different matter.

> Part of me wants to guide our users to write a custom error handler that would "fail fast", giving them some control of what to ignore and what to address, however that could be daunting for them in production environments when sifting through errors.

I think that's good guidance.

> That lead me down the path of reporting errors using `OpenTelemetry.handle_error`. If it is indeed our preference to Fail-Fast when it comes to Exporter configurations then I will go back to the fail-fast approach and raise `ArgumentError` instead.

Any errors raised during configuration are passed to the error handler anyway, so we should simply `raise` and let `configure` handle the error: https://github.com/open-telemetry/opentelemetry-ruby/blob/666b81204fc5d0a34744d9e88788127897becd01/sdk/lib/opentelemetry/sdk.rb#L71

Reply:
:+1: `raise hell`

Reply:
I drafted something but gotta say. I don't love it. Need to sleep on this.